### ISSUE TITLE
"Initializing JS/TS language features" continues indefinitely and hogs system CPU

### ISSUE THREAD

**Issue by damms005** [microsoft/vscode#87914]
- VSCode Version: 1.41.1
- OS Version: Ubuntu 18.04

Steps to Reproduce:

1. Open vscode
2. Open project folder

Does this issue occur when all extensions are disabled?: Sometimes, depending on size of folder, if project contains both PHP files and JavaScript files

My request: 
I will prefer to have a working vscode editor that this JS/TS bringing things to halt. How do I completely disable JS/TS features, so that vscode does not even bother to  initialize JS/TS language features at all in the first place?

Thanks

**Comment by EdwardDiehl** [comment]
Same here

- VSCode Version: 1.41.1
- OS Version: Ubuntu 16.04

It loads my laptop CPU and consumes quite a lot of memory continuously. Even when I close vscode the processes are still there. The issue does not occur on small size projects.
The issue was also occurring in several previous versions of vscode.

Here is a screenshot with some of processes in htop.

![htop-vscode](https://user-images.githubusercontent.com/657395/71626467-04c21980-2bf6-11ea-9ae3-f74860bedd61.png)

**Comment by damms005** [comment]
Something about `tsserver.log` that may make some sense:

```
Info 16   [8:22:1.542] Open files: 
Info 16   [8:22:1.542] Structure after ensureProjectForOpenFiles:
Info 17   [8:22:1.542] Open files: 
Info 17   [8:22:1.542] got projects updated in background, updating diagnostics for 
Info 18   [8:36:15.410] request:
    {"seq":2,"type":"request","command":"updateOpen","arguments":{"changedFiles":[],"closedFiles":[],"openFiles":[<files and their contents go here>]}}
Info 19   [8:36:15.412] Search path: /opt/lampp/htdocs/hidden-laravel-backends/telegramfx
Info 20   [8:36:15.413] ConfigFilePresence:: Current Watches: :: File: /my/custom/private/project/full/path/tsconfig.json Currently impacted open files: RootsOfInferredProjects:  OtherOpenFiles: /my/custom/private/project/full/path/webpack.mix.js Status: File added to open files impacted by this config file
Info 21   [8:36:15.413] ConfigFilePresence:: Current Watches: :: File: /my/custom/private/project/full/path/jsconfig.json Currently impacted open files: RootsOfInferredProjects:  OtherOpenFiles: /my/custom/private/project/full/path/webpack.mix.js Status: File added to open files impacted by this config file
Info 22   [8:36:15.413] For info: /my/custom/private/project/full/path/webpack.mix.js :: No config files found.

```
Notice how it takes almost 15 minutes to go from the `Info 17` line to `Info 18`?

**Comment by jarretmoses** [comment]
I can confirm this happening to me as well. Here is a gist of what my logs output/ https://gist.github.com/jarretmoses/12861f3a299dad9859c196784afde453. You can see the errors toward the end

**Comment by mjbvz** [comment]
@damms005 Please follow [these instructions](https://github.com/Microsoft/vscode/wiki/Performance-Issues) to collect performance information. Share the output of `code --status`

**Comment by damms005** [comment]
@mjbvz 

The output of `code --status`:

```
[25252:0108/001056.076568:ERROR:sandbox_linux.cc(369)] InitializeSandbox() called with multiple threads in process gpu-process.
Version:          Code 1.41.1 (26076a4de974ead31f97692a0d32f90d735645c0, 2019-12-18T15:04:31.999Z)
OS Version:       Linux x64 4.15.0-72-generic
CPUs:             Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 2355)
Memory (System):  15.41GB (0.31GB free)
Load (avg):       11, 7, 6
VM:               0%
Screen Reader:    no
Process Argv:     --no-sandbox --unity-launch
GPU Status:       2d_canvas:                     enabled
                  flash_3d:                      enabled
                  flash_stage3d:                 enabled
                  flash_stage3d_baseline:        enabled
                  gpu_compositing:               enabled
                  metal:                         disabled_off
                  multiple_raster_threads:       enabled_on
                  oop_rasterization:             disabled_off
                  protected_video_decode:        unavailable_off
                  rasterization:                 disabled_software
                  skia_renderer:                 disabled_off
                  surface_control:               disabled_off
                  surface_synchronization:       enabled_on
                  video_decode:                  unavailable_off
                  viz_display_compositor:        enabled_on
                  viz_hit_test_surface_layer:    disabled_off
                  webgl:                         enabled
                  webgl2:                        enabled

CPU %	Mem MB	   PID	Process
    0	    79	 17755	code main
    0	     0	 17762	   zygote
    4	    47	 17794	   gpu-process
    1	   237	 17860	   window (/opt/lampp/htdocs/hidden-laravel-backends/schoolserver/resources/views/agents/register.blade.php - Visual Studio Code)
    0	   110	 18076	     extensionHost
    0	    32	 18174	       electron_node tsserver.js 
    2	  1184	 18175	       electron_node tsserver.js 
    0	    32	 18191	         electron_node typingsInstaller.js typesMap.js 
    0	    47	 18266	       electron_node htmlServerMain.js 
    0	   110	 18296	       electron_node server.js 
    0	   316	 19142	       electron_node intelephense.js 
    0	   189	 18100	     watcherService
    0	    32	 18148	     searchService
    0	    63	 18033	   shared-process
    0	     0	 25288	     /bin/sh -c /bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	     0	 25289	       /bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	    79	 23970	   window (Issue Reporter)

Workspace Stats: 
|  Window (/opt/lampp/htdocs/hidden-laravel-backends/schoolserver/resources/views/agents/register.blade.php - Visual Studio Code)
|    Folder (schoolserver): more than 30137 files
|      File types: php(6750) js(893) json(644) jpg(638) svg(582) css(565)
|                  png(530) md(519) html(300) less(269)
|      Conf files: package.json(20) makefile(7) grunt.js(3) launch.json(1)
|                  settings.json(1) gulp.js(1) tsconfig.json(1)
|                  webpack.config.js(1)
|      Launch Configs: php(2)

```

**Comment by damms005** [comment]
@mjbvz  On another random occasion, it happened again and I followed the instruction in the link you sent, and I have this file: 
[CPU-20200108T000905.816Z.cpuprofile.txt](https://github.com/microsoft/vscode/files/4033038/CPU-20200108T000905.816Z.cpuprofile.txt)

Now I tried reproducing it and I can *100% confirm that it always happen only when I open a JavaScript file (i.e. any .js gile)*

So whenever I open a new workspace/folder and no js file is opened in the editor then everything is calm and cool. The moment I open a .js file, all hell break loose withing seconds!!

**Comment by mjbvz** [comment]
Can you make sure you run `--status` when the high cpu usage is happening? The current log doesn't show high cpu usage 

Also please try upgrading your workspace to use `typescript@next` by installing [this extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next). Does that fix the problem

**Comment by damms005** [comment]
After monitoring the behavior for some time now, I noticed that the "high CPU usage" is not synonymous with "vscode is freezing and unresponsive". The profile log that I attached previously happened when vscode froze, albeit every other app in other windows I was working on at the moment were responding (although they were slow-ish, but the picked-up again anytime vscode unfreezes).

Also, the typescript@next extension did not do have any effect on the situation.

I want to suggest that the "hotfix" for this very common and rampant issue (which is easily confirmed by doing a quick search in this repo's github issues although likely by diff causes) is to *disable the whole inbuilt typescript support or pipeline whatever*!

This is much so because it is not a third-party extension thing. Why is something that irrelevant to a good number of people giving so much headache? I don't use typescript and I don't plan to use it anytime soon. Might be a bad decision but yet it is suppose to be a choice. Worse still is that the whole typescript thing is tied to JavaScript, so disabling that entire typescript inbuilt whatever may cause JavaScript stuff to stop working, but I can bet you if you put that to vote a whole lot of people will prefer that to this current headache.

LET VSCODE ALLOW DEVS TO DISABLE TYPESCRIPT IF THEY SO CHOOSE!!! IT IS GOOD DX TO LET DEVS CHOOSE THEIR SOURCE OF HEADACHES!!!
